relayer/relayer.go

```diff
@@ -149,7 +150,10 @@ func NewRelayer(
 			)
 			return nil, nil, err
 		}
-		sub.ProcessFromHeight(big.NewInt(0).SetUint64(height))
+		// Process historical blocks asynchronously so that the main processing loop can
+		// start processing new blocks as soon as possible. Otherwise, it's possible for
+		// ProcessFromHeight to overload the message queue and cause a deadlock.
+		go sub.ProcessFromHeight(big.NewInt(0).SetUint64(height), doneProcessingMissedBlocks)
```

Comment:
So we have an issue here where all errors from this function now get swallowed. If you leave a TODO here, I'll plumb through the context that I implemented in the failure recovery PR, which will allow us to both cancel the context from inside the goroutine, and listen for cancellations of the other goroutines.

Reply:
As mentioned [here](https://github.com/ava-labs/awm-relayer/pull/140#discussion_r1456613267), I repurposed the bool channel to also pass an error to the select loop in `ProcessLogs`. This could be done with an atomic bool and the context, but using the channel is more consistent with how we're treating async errors elsewhere, such as the subscriber.